sqlmesh/utils/cron.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+import typing as t
+
+from croniter import croniter
+
+from sqlmesh.utils import ttl_cache
+from sqlmesh.utils.date import TimeLike, now, to_datetime
+
+
+@ttl_cache(maxsize=1280000)
+def cron_next(cron: str, time: TimeLike) -> float:
+    return croniter(cron, to_datetime(time)).get_next()
+
+
+@ttl_cache(maxsize=1280000)
+def cron_prev(cron: str, time: TimeLike) -> float:
+    return croniter(cron, to_datetime(time)).get_prev()
```

Comment:
why aren't we better off caching the `croniter` instance itself?

Reply:
cache works on methods, also ceo inter is stateful